Source/WebKit/Platform/ios/VideoPresentationInterfaceLMK.mm

```diff
@@ -318,13 +340,20 @@ - (void)layoutSublayers
 
     ALWAYS_LOG_IF_POSSIBLE(LOGIDENTIFIER);
     m_playerViewController = [linearMediaPlayer() makeViewController];
-    [m_playerViewController viewController].view.alpha = 0;
+    [m_playerViewController.get() viewController].view.alpha = 0;
+
+    if (RefPtr model = videoPresentationModel()) {
+        if (CheckedPtr playbackSessionModel = playbackSessionInterface().playbackSessionModel())
+            [m_playerViewController setPrefersAutoDimming:playbackSessionModel->prefersAutoDimming()];
+    }
+
 }
 
 void VideoPresentationInterfaceLMK::swapFullscreenModesWith(VideoPresentationInterfaceIOS& otherInterfaceIOS)
 {
     auto& otherInterface = static_cast<VideoPresentationInterfaceLMK&>(otherInterfaceIOS);
     std::swap(m_playerViewController, otherInterface.m_playerViewController);
+    std::swap(m_savedAutoDimmingPreference, otherInterface.m_savedAutoDimmingPreference);
```

Comment:
as discussed, I don't think we want to use std::swap here, should just copy the value